src/test/java/mate/academy/bookshop/repository/book/BookRepositoryTest.java

```diff
@@ -0,0 +1,172 @@
+    private Book book1;
+    private Book book2;
+    private Book book3;
+
+    @BeforeEach
+    void setUp() {
+        category = new Category();
+        category.setName("Category1");
+        category = categoryRepository.save(category);
+
+        book1 = createBook("Title1", "234567", category);
+        book2 = createBook("Title2", "234560", category);
+        book3 = createBook("Title3", "234599", category);
+    }
+
+    @DisplayName("Find all with default pageable params")
+    @Test
+    void findAll_PageableDefaultParams_ReturnsPageOfBooks() {
+        Pageable pageable = PageRequest.of(0, 20);
+        List<Book> books = List.of(book1, book2, book3);
+        bookRepository.saveAll(books);
+        Page<Book> bookPage = new PageImpl<>(books, pageable, 3);
+
+        Page<Book> response = bookRepository.findAll(pageable);
+
+        Assertions.assertEquals(20, response.getSize());
+        Assertions.assertEquals(3, response.getTotalElements());
+        Assertions.assertEquals(bookPage, response);
+        Assertions.assertTrue(response.getContent().containsAll(books));
+    }
+
+    @DisplayName("Find all with custom pageable params")
+    @Test
+    void findAll_PageableCustomParams_ReturnsPageOfBooks() {
+        Pageable pageable = PageRequest.of(0, 2, Sort.by("title").descending());
+        List<Book> books = List.of(book1, book2, book3);
+        Page<Book> bookPage = new PageImpl<>(List.of(book3, book2), pageable, 3);
+        bookRepository.saveAll(books);
+
+        Page<Book> response = bookRepository.findAll(pageable);
+
+        Assertions.assertEquals(2, response.getSize());
+        Assertions.assertEquals(List.of(book3, book2), response.getContent());
+    }
+
+    @DisplayName("Find by id when book exists")
+    @Test
+    void findById_BookExistById_ReturnsOptionalBookWithCategories() {
+        Book savedBook = bookRepository.save(book1);
+
+        Optional<Book> response = bookRepository.findById(savedBook.getId());
+
+        Assertions.assertTrue(response.isPresent());
+        Book fetchedBook = response.get();
+        Assertions.assertEquals(1, fetchedBook.getCategories().size());
+        Assertions.assertTrue(fetchedBook.getCategories().contains(category));
+    }
+
+    @DisplayName("Find by id when book does not exist")
+    @Test
+    void findById_BookNotExistById_ReturnsOptionalEmpty() {
+        Optional<Book> response = bookRepository.findById(1L);
+
+        Assertions.assertFalse(response.isPresent());
+        Assertions.assertEquals(Optional.empty(), response);
+    }
+
+    @DisplayName("Find by category id with valid category id and default pageable params")
+    @Test
+    void findByCategoryId_ValidCategoryIdAndDefaultPageableParams_ReturnsPageOfBooks() {
+        Category category2 = new Category();
+        category2.setName("Category2");
+        category2.setDescription("Description2");
+        Category secondCategory = categoryRepository.save(category2);
```

Comment:
Sometimes you extract logic of mock object creation to method, sometimes not, please be consistent

Reply:
Not resolved